ndb_api/db_helpers.go

```diff
@@ -142,4 +115,254 @@ func GenerateDeprovisionDatabaseRequest() (req *DatabaseDeprovisionRequest) {
 		DeleteLogicalCluster: true,
 	}
 	return
+
+}
+
+func validateReqData(ctx context.Context, databaseInstanceType string, reqData map[string]interface{}) (err error) {
+	log := ctrllog.FromContext(ctx)
+	dbPassword, ok := reqData[common.NDB_PARAM_PASSWORD].(string)
+	// Type Assertion for dbPassword
+	if !ok || dbPassword == "" {
+		err = errors.New("invalid database password")
+		var errStatement string
+		if !ok {
+			errStatement = "Type assertion failed for database password. Expected a string value"
+		} else {
+			errStatement = "Empty database password"
+		}
+		log.Error(err, errStatement)
+		return
+	}
+
+	// Type Assertion for SSHKey
+	if databaseInstanceType != common.DATABASE_TYPE_MSSQL {
+		SSHPublicKey, ok := reqData[common.NDB_PARAM_SSH_PUBLIC_KEY].(string)
+		if !ok || SSHPublicKey == "" {
+			err = errors.New("invalid ssh public key")
+			var errStatement string
+			if !ok {
+				errStatement = "Type assertion failed for SSHPublicKey. Expected a string value"
+			} else {
+				errStatement = "Empty SSHPublicKey"
+			}
+			log.Error(err, errStatement)
+			return
+		}
+	}
+	return
+}
+
+// Appends request based on database type
+type DBProvisionRequestAppender interface {
+	appendRequest(req *DatabaseProvisionRequest, database DatabaseInterface, reqData map[string]interface{}) *DatabaseProvisionRequest
+}
+
+type MSSQLProvisionRequestAppender struct{}
+
+type MongoDbProvisionRequestAppender struct{}
+
+type PostgresProvisionRequestAppender struct{}
+
+type MySqlProvisionRequestAppender struct{}
+
+func (a *MSSQLProvisionRequestAppender) appendRequest(req *DatabaseProvisionRequest, database DatabaseInterface, reqData map[string]interface{}) *DatabaseProvisionRequest {
```

Comment:
Should we also return an error from the `appendRequest` function?

Reply:
Yes if we add the type assertion validations, we would need to